api/wharfapi/v5/builds.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+package wharf.api.v5;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/iver-wharf/wharf-api/v5/api/wharfapi/v5";
+
+// Builds contains Wharf build-oriented RPCs.
+service Builds {
+  // CreateLogStream allows creating logs as a client-side stream.
+  // Logs targeting non-existing builds as well as logs that has already been
+  // added before (based on the build, log, and step IDs) will be discarded.
+  rpc CreateLogStream(stream CreateLogStreamRequest)
+    returns (CreateLogStreamResponse);
+}
+
+// CreateLogStreamRequest contains the streamed log lines that meant to be
+// created.
+message CreateLogStreamRequest {
+  // Lines is an array of log lines to allow chunking log line streams.
+  repeated NewLogLine lines = 1;
+}
+
+// CreateLogStreamResponse is the response returned after closing a log creation
+// stream.
+message CreateLogStreamResponse {
+  // LinesInserted is the number of lines that has been inserted in total by
+  // this stream.
+  uint64 lines_inserted = 1;
+}
+
+// NewLogLine is a single log line produced by a Wharf build to be added to the
+// database.
+message NewLogLine {
+  // BuildId is the database ID of the build this log line belongs to.
+  uint64 build_id = 1;
+  // LogId is the worker's own ID of the log line. It's unique per build step
+  // for a given build, but may have collisons across multiple steps or builds.
+  uint64 log_id = 2; // not DB value; worker's own ID of the log line
+  // StepId is the worker's own ID of the step. It's unique for a given build,
+  // but may have collisons across multiple builds.
+  uint64 step_id = 3; // not DB value; worker's own ID of the step ID
```

Comment:
Nit: I don't like that we're being forced by this one file to have inconsistent naming in our Go code.
I would prefer we break convention in proto and get `XxxID` for our Go code, like:

```suggestion
  // BuildID is the database ID of the build this log line belongs to.
  uint64 buildID = 1;
  // LogID is the worker's own ID of the log line. It's unique per build step
  // for a given build, but may have collisions across multiple steps or builds.
  uint64 logID = 2; // not DB value; worker's own ID of the log line
  // StepID is the worker's own ID of the step. It's unique for a given build,
  // but may have collisions across multiple builds.
  uint64 stepID = 3; // not DB value; worker's own ID of the step ID
```

Reply:
Actually kinda changed my mind, better to follow convention for proto so it's easier to maintain consistency between protofiles perhaps.

Maybe there are some tags/other ways to change the generated name though?

Reply:
I'm following the official style guide here https://developers.google.com/protocol-buffers/docs/style#message_and_field_names

I did some digging and found this: https://github.com/alta/protopatch

I've updated this and the wharf-api-client-go PR to make use of it.

I did try out https://github.com/gogo/protobuf but that project is both dying and has lacking support for Protobuf v3. This alta/protopatch solution is only augmenting the official Go protobuf generator instead of replacing it.

Reply:
Neat 👍🏻